build-system/tasks/bundle-size.js

```diff
@@ -137,40 +196,56 @@ async function checkBundleSize() {
   }
 
   const maxSize = await getMaxBundleSize();
+  const ancestorBundleSize = await getAncestorBundleSize();
 
-  const cmd = `npx bundlesize -f "${runtimeFile}" -s "${maxSize}"`;
-  log('Running ' + cyan(cmd) + '...');
-  const output = getStdout(cmd);
-  const pass = output.match(/PASS .*/);
-  const fail = output.match(/FAIL .*/);
-  const error = output.match(/ERROR .*/);
-  const bundleSizeFormatMatches = output.match(/: (\d+.?\d*KB)/);
-  if (error && error.length > 0) {
-    log(yellow(error[0]));
-    process.exitCode = 1;
-    return;
-  } else if (!bundleSizeFormatMatches) {
-    log(red('ERROR:'), 'could not infer bundle size from output.');
-    log(yellow(output));
-    process.exitCode = 1;
-    return;
-  } else if (fail && fail.length > 0) {
-    log(red(fail[0]));
-    log(red('ERROR:'), cyan('bundlesize'), red('found that'),
-        cyan(runtimeFile), red('has exceeded its size cap of'),
-        cyan(maxSize) + red('.'));
-    log(red(
-        'This is part of a new effort to reduce AMP\'s binary size (#14392).'));
-    log(red('Please contact @choumx or @jridgewell for assistance.'));
-    process.exitCode = 1;
-  } else if (pass && pass.length > 0) {
-    log(green(pass[0]));
-  } else {
-    log(yellow(output));
+  let compareAgainstMaxSize = true;
+  let output, status, newBundleSize;
+  if (ancestorBundleSize) {
+    [output, status, newBundleSize] = compareBundleSize(ancestorBundleSize);
+    switch (status) {
+      case STATUS_ERROR:
+        log(yellow(output));
+        process.exitCode = 1;
+        return;
+      case STATUS_FAIL:
+        log(yellow('New bundle size of'), cyan(newBundleSize),
+            yellow('is larger than the ancestor\'s bundle size of'),
+            cyan(ancestorBundleSize));
```

Comment:
It'd be nice to print the size delta here. I always forget to carry the one. `#jk` `#notactually`

Reply:
Done, added text at the end of the failure message, e.g., " (Δ +0.05KB)"